### PR TITLE
gh-121245: Amend d611c4c8e9 (correct import)

### PR DESCRIPTION
**Pull request by skirpichev**
This prevents writing incorrect history (empty).

Fixed by @domdfcoding.

* Issue: gh-121245

### CODE REVIEW

**Comment by skirpichev** [comment]
CC @pablogsal

File: Lib/site.py [hunk 1]
```diff
@@ -527,7 +527,7 @@ def register_readline():
         def write_history():
             try:
                 # _pyrepl.__main__ is executed as the __main__ module
```
Comment (vstinner) [inline Lib/site.py]: Either update or remove the comment.

**Review by vstinner — APPROVED** [review]
LGTM if you remove the comment.

File: Lib/site.py (hunk 1 above)
Comment (skirpichev) [inline Lib/site.py]:
```suggestion
```

**Comment by skirpichev** [comment]
BTW, test in write_history() looks asymmetrical wrt reading the history file.  Maybe it should be moved up to fill some variable like ``using_base_repl``?

That wasn't included, I did just minimal changes.

File: Misc/NEWS.d/next/Library/2024-07-02-11-34-06.gh-issue-121245.sSkDAr.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Correct working with CLI history for new REPL.
```
Comment (hroncok) [inline Misc/NEWS.d/next/Library/2024-07-02-11-34-06.gh-issue-121245.sSkDAr.rst]:
```suggestion
Fix a bug in the handling of the command history of the new :term:`REPL` that caused
the history file to be wiped at REPL exit.
```

**Review by hroncok — APPROVED** [review]

**Pull request merged**